Admin/Client/src/App.tsx

```diff
@@ -1,7 +1,57 @@
-import React from 'react';
-
-function App() {
-    return <div className='App'></div>;
-}
-
-export default App;
+import React, { useState } from 'react';
+import InitSideEffects from './sideEffects/initSideEffects';
+import LoginPage from './pages/loginPage';
+import Authentication from './authentication';
+
+import {
+    BrowserRouter as Router,
+    Switch,
+    Route,
+    Link,

+    Redirect
+} from "react-router-dom";
+import MainPage from './pages/mainPage';
+
+//Change this to move from real to mock side effects
+const isMocking = true;
+InitSideEffects(isMocking);
+
+function App() {
+    const [isLoggedin, setLoginStatus] = useState(false);

```

Comment:
maybe use cookie?
since keeping state in the top level components may cause extra renders no?

Reply:
When it changes you should render child right? otherwise nothing will happen - we don't have redux here. 
Do you suggest something else?